datadog_checks_base/datadog_checks/base/checks/windows/perf_counters/base.py

```diff
@@ -64,7 +63,7 @@ def _query_counters(self):
         # https://docs.microsoft.com/en-us/windows/win32/api/pdh/nf-pdh-pdhenumobjectitemsa#remarks
 
         if self.should_refresh_enum_obj_list():
-            with self.enum_object_list_lock.write_lock():
+            with self.nonblocking_lock():
```

Comment:
@yzhan289  How would it prevent double sequential execution

Reply:
Sorry for the delay! My thinking was the following: if there were multiple instances running this check and should refresh enum object list, then only one of the instances is allowed to do it while the other instances would just skip this section. Would double-checked locking be more suitable than a nonblocking lock?